app/src/main/java/com/github/orkest/View/profile/ProfileTopInterface.kt

```diff
@@ -157,23 +158,11 @@ fun FollowButton(viewModel: ProfileViewModel, isUserFollowed: Boolean?){
 }
 
 private fun followOrUnfollow(viewModel: ProfileViewModel, isUserFollowed: Boolean) {
-    if (isUserFollowed) {
-        //The unfollow button is displayed at this stage, when clicked on,
-        //the current user unfollows the account and isUserFollowed is updated to false
-        viewModel.unfollow().whenComplete { _, _ ->
-            viewModel.isUserFollowed.value = false
-        }
-    } else {
-        //The follow button is displayed at this stage, when clicked on,
-        //the current user follows the account and isUserFollowed is updated to true
-        viewModel.follow().whenComplete { _, _ ->
-            viewModel.isUserFollowed.value = true
-        }
+    CompletableFuture.allOf(viewModel.updateUserFollowers(!isUserFollowed), viewModel.updateCurrentUserFollowings(!isUserFollowed)).thenApply{
+        viewModel.isUserFollowed.value = !isUserFollowed
     }
 }

```

Comment:
Maybe handle the exception with something like :

.exceptionally { exception ->
        // Handle the exception here
        null
    }
    
   In case there are unexpected behaviors